src/pytorch_lightning/utilities/data.py

```diff
@@ -126,7 +126,7 @@ def has_len(dataloader: Union[DataLoader, Iterable]) -> bool:
 
 
 def has_len_all_ranks(
-    dataloader: DataLoader,
+    dataloader: Any,
```

Comment:
shall this be at least an Iterable type?

Reply:
Thanks for your review and comment. After I tried some experiments, I found that this modification is unnecessary for this PR. So I reverted this line.